src/Item/Item.router.spec.ts

```diff
@@ -278,41 +324,97 @@ describe('Item router', () => {
+        )
+        ;(Collection.findOne as jest.Mock).mockResolvedValueOnce(
+          dbCollectionMock
+        )
+        ;(Collection.findByIds as jest.Mock).mockResolvedValueOnce([
+          dbCollectionMock,
+        ])
+        mockItemConsolidation([dbItemMock], [wearable])
+        url = `/collections/${dbCollectionMock.id}/items`
+      })
+
+      it('should return all the items of a collection that are published with URN and the ones that are not without it', () => {
+        return server
+          .get(buildURL(url))
+          .set(createAuthHeaders('get', url))
+          .expect(200)
+          .then((response: any) => {
+            expect(response.body).toEqual({
+              data: [
+                {
+                  ...resultingItem,
+                  beneficiary: itemFragment.beneficiary,
+                  collection_id: dbItem.collection_id,
+                  blockchain_item_id: dbItem.blockchain_item_id,
+                  urn,
+                },
+                resultItemNotPublished,
+              ],
+              ok: true,
+            })
+          })
+      })
     })
-    it('should return all the items of a collection that are published with URN and the ones that are not without it', () => {
-      return server
-        .get(buildURL(url))
-        .set(createAuthHeaders('get', url))
-        .expect(200)
-        .then((response: any) => {
-          expect(response.body).toEqual({
-            data: [
-              {
-                ...resultingItem,
-                beneficiary: itemFragment.beneficiary,
-                collection_id: dbItem.collection_id,
-                blockchain_item_id: dbItem.blockchain_item_id,
-                urn,
-              },
-              resultItemNotPublished,
-            ],
-            ok: true,
+
+    describe('and the collection is a third party collection', () => {
+      beforeEach(() => {
+        ;(Item.find as jest.Mock).mockResolvedValueOnce([
+          dbTPItemMock,
+          dbItemNotPublished,
+        ])
+        ;(collectionAPI.fetchCollectionWithItemsByContractAddress as jest.Mock).mockResolvedValueOnce(
+          { collection: itemFragment.collection, items: [itemFragment] }
+        )
+        ;(Collection.findOne as jest.Mock).mockResolvedValueOnce(
+          dbTPCollectionMock
+        )
+        ;(Collection.findByIds as jest.Mock).mockResolvedValueOnce([
+          dbTPCollectionMock,
+        ])
+        ;(thirdPartyAPI.fetchLastItem as jest.Mock).mockResolvedValueOnce(
+          thirdPartyItemFragmentMock
+        )
+        ;(thirdPartyAPI.fetchItemsByCollection as jest.Mock).mockResolvedValueOnce(
+          [thirdPartyItemFragmentMock]
+        )
+        mockItemConsolidation([dbItemMock], [wearable])
+        url = `/collections/${dbCollectionMock.id}/items`
+      })
+
+      it('should return all the items of a collection that are published with URN and the ones that are not without it', () => {
```

Comment:
As the TP items must contain the URN:
```suggestion
      it('should return all the items of a collection that are published with their URN, () => {
```